src/qt/walletmodel.cpp

```diff
@@ -397,25 +394,27 @@ WalletModel::SendCoinsReturn WalletModel::sendCoins(WalletModelTransaction &tran
                 }
 
                 // Store PaymentRequests in wtx.vOrderForm in wallet.
-                std::string key("PaymentRequest");
                 std::string value;
                 rcp.paymentRequest.SerializeToString(&value);
-                newTx->vOrderForm.push_back(make_pair(key, value));
+                vOrderForm.emplace_back("PaymentRequest", std::move(value));
             }
             else if (!rcp.message.isEmpty()) // Message from normal dash:URI (dash:XyZ...?message=example)
-            {
-                newTx->vOrderForm.push_back(make_pair("Message", rcp.message.toStdString()));
-            }
+                vOrderForm.emplace_back("Message", rcp.message.toStdString());
+        }
+
+        if (fIsPrivateSend) {
+            vOrderForm.emplace_back("DS", "1");
         }
 
+        CTransactionRef& newTx = transaction.getTransaction();
         CReserveKey *keyChange = transaction.getPossibleKeyChange();
         CValidationState state;
-        if(!wallet->CommitTransaction(*newTx, *keyChange, g_connman.get(), state))
+        if (!wallet->CommitTransaction(newTx, {} /* mapValue */, std::move(vOrderForm), {} /* fromAccount */, *keyChange, g_connman.get(), state))
             return SendCoinsReturn(TransactionCommitFailed, QString::fromStdString(state.GetRejectReason()));
 
         CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
-        ssTx << *newTx->tx;
-        transaction_array.append(ssTx.data(), ssTx.size());
+        ssTx << newTx;
+        transaction_array.append(&(ssTx[0]), ssTx.size());
```

Comment:
This line wasn't touched in the original PR, should be reverted
```suggestion
        transaction_array.append(ssTx.data(), ssTx.size());
```

Reply:
Hm right. Seems like this line wasn't upstream before because you made it diverge from it here https://github.com/dashpay/dash/pull/3276/commits/5f8bcacd7fa15b411e6540a7102e25ca68a3453f, now please feel a little bad for a moment 😄  

There the original PR did not include the 

```
-        transaction_array.append(&(ssTx[0]), ssTx.size());
+        transaction_array.append(ssTx.data(), ssTx.size());
```
Bitcoin `master` still has old way there https://github.com/bitcoin/bitcoin/blob/master/src/qt/walletmodel.cpp#L247

So, im just curious.. would it be bad some reason to apply such changes still even if they are related to other PRs? Just thinking if situations like this could be good opportunities to get a bit closer to upstream here and there.

Reply:
Ok no actually to apply this to this PR doesn't make any sense at all.. guess that would need to be a "revert some stuff" PR then to get such things to upstream again.

Reply:
this ^^^ or you could fix upstream instead :)